Update timeout for force update , add 30 sec interval to wait network

diff --git a/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java b/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
index 0e48a43..6be3ee2 100644
--- a/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
+++ b/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
@@ -551,7 +551,11 @@ private static void scheduleUpdate(Context context, long timeFromNow, boolean fo
     public static void scheduleNextUpdate(Context context, boolean force) {
         if (force) {
             if (D) Log.d(TAG, "Scheduling next update immediately");
-            scheduleUpdate(context, 0, true);
+            if (WidgetUtils.isNetworkAvailable(context)) {
+                scheduleUpdate(context, 0, true);
+            } else {
+                scheduleUpdate(context, 30000, true); // add 30 sec interval to wait network
+            }
         } else {
             final long lastUpdate = Preferences.lastWeatherUpdateTimestamp(context);
             final long interval = Preferences.weatherRefreshIntervalInMs(context);
